src/cobalt_model/pagination_config.rs

```diff
@@ -1,108 +1,109 @@
-use liquid;
-use std::borrow::Cow;
+use std::convert::Into;
 use std::vec::Vec;
+
+use liquid;
+
 use super::SortOrder;
 
-// struct Trails {
-//   before: i32,
-//   after: i32,
-// }
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq)]
+#[serde(deny_unknown_fields)]
+pub enum Include {
+  None,
+  #[serde(rename = "all")]
+  All,
+  // tags,
+}
+
+impl Into<liquid::Scalar> for Include {
+  fn into(self) -> liquid::Scalar {
+    match self {
+      Include::None => liquid::Scalar::new(""),
+      Include::All => liquid::Scalar::new("all"),
+    }
+  }
+}
 
-#[derive(Clone, Debug)]
-pub(crate) struct PaginationCfg {
-  pub(crate) include: String,
-  pub(crate) per_page: i32,
-  pub(crate) permalink: String,
-  pub(crate) order: SortOrder,
-  pub(crate) sort_by: Vec<String>,
-  pub(crate) post_include: Option<String>
-  //trails: Trails,
+impl Default for Include {
+  fn default() -> Include {
+    Include::None
+  }
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq)]
+#[serde(deny_unknown_fields, default)]
+pub struct PaginationConfig {
```

Comment:
Somewhere between this and the frontmatter, we are not handling the shortcuts.

Reply:
On second thought, I'm wanting to keep this PR as small as possible to enable easier reviewing.  I'd prefer if shortcuts and other types of pagination are explicitly listed in the PR as not included and that we handle those in separate PRs.

Reply:
I mostly agree, but I might need to have "tags" (if we keep it, as it is under discussion in the RFC) as it helps me to see if my code can handle more generic cases. The `all` case is quite simple, but when I started to code `tags` I had to change some things around.